src/core/thread/mlr_manager.cpp

```diff
@@ -403,6 +496,43 @@ void MlrManager::HandleMulticastListenerRegistrationResponse(Coap::Message *
     }
 }
 
+otError MlrManager::ParseMulticastListenerRegistrationResponse(otError        aResult,
+                                                               Coap::Message *aMessage,
+                                                               uint8_t &      aStatus,
+                                                               Ip6::Address * aFailedAddresses,
+                                                               uint8_t &      aFailedAddressNum)
+{
+    otError  error;
+    uint16_t addressesOffset, addressesLength;
+
+    aStatus = ThreadStatusTlv::MlrStatus::kMlrGeneralFailure;
+
+    VerifyOrExit(aResult == OT_ERROR_NONE && aMessage != nullptr, error = OT_ERROR_PARSE);
+    VerifyOrExit(aMessage->GetCode() == OT_COAP_CODE_CHANGED, error = OT_ERROR_PARSE);
+
+    SuccessOrExit(error = Tlv::FindUint8Tlv(*aMessage, ThreadTlv::kStatus, aStatus));
+
+    if (ThreadTlv::FindTlvValueOffset(*aMessage, IPv6AddressesTlv::kIPv6Addresses, addressesOffset, addressesLength) ==
+        OT_ERROR_NONE)
+    {
+        VerifyOrExit(addressesLength % sizeof(Ip6::Address) == 0, error = OT_ERROR_PARSE);
+        VerifyOrExit(addressesLength / sizeof(Ip6::Address) <= kIPv6AddressesNumMax, error = OT_ERROR_PARSE);
+
+        for (uint16_t offset = 0; offset < addressesLength; offset += sizeof(Ip6::Address))
+        {
+            IgnoreReturnValue(aMessage->Read(addressesOffset + offset, sizeof(Ip6::Address),
```

Comment:
ditto.

Reply:
It should be 
```
IgnoreReturnValue(
                aMessage->Read(addressesOffset + offset, sizeof(Ip6::Address), &aFailedAddresses[aFailedAddressNum]));
```

Note that it can't fail here.